ShimmerAndroidInstrumentDriver/shimmerServiceExample/src/main/java/com/shimmerresearch/shimmerserviceexample/MainActivity.java

```diff
@@ -305,6 +305,10 @@ public void handleMessage(Message msg) {
                         }
                         break;
                     case SDLOGGING:
+                        List<ShimmerDevice> deviceList2 = mService.getListOfConnectedDevices();
+                        if(deviceList2.size() > connectedShimmersListFragment.getNumShimmersConnected()) {
```

Comment:
to be more comprehensive lets not have the size check, there might be use cases we aren't thinking off